solver/llbsolver/bridge.go

```diff
@@ -158,17 +161,53 @@ func (b *llbBridge) Solve(ctx context.Context, req frontend.SolveRequest, sid st
 		return &frontend.Result{}, nil
 	}
 
+	if res.Metadata == nil {
+		res.Metadata = make(map[string][]byte)
+	}
+
+	metadata := make(map[string][]byte)
+	for k, v := range res.Metadata {
+		if !strings.HasPrefix(k, exptypes.ExporterBuildInfo) {
+			metadata[k] = v
+			continue
+		}
+		bi := &binfotypes.BuildInfo{}
+		if errm := json.Unmarshal(v, bi); errm != nil {
+			return nil, errors.Wrapf(errm, "failed to unmarshal build info for %q", k)
+		}
+		if bi.Frontend == "" {
+			bi.Frontend = req.Frontend
+		}
+		bi.Attrs = buildinfo.ReduceMap(req.FrontendOpt, bi.Attrs)
+		dtbi, errm := json.Marshal(bi)
+		if errm != nil {
+			return nil, errm
+		}
+		metadata[k] = dtbi
+	}
+	if _, ok := metadata[exptypes.ExporterBuildInfo]; !ok {
```

Comment:
if image is multi-platform doesn't it also end up in here as there is no `exptypes.ExporterBuildInfo` key (there is `exptypes.ExporterBuildInfo` + `platform`). 

And if the result is multi-platform but there is no metadata at all, shouldn't metadata be created here for each platform?

Reply:
> if image is multi-platform doesn't it also end up in here as there is no `exptypes.ExporterBuildInfo` key (there is `exptypes.ExporterBuildInfo` + `platform`).

I will take another look on that one but I recall I needed it also for multi-platform otherwise attributes are lost but I might be wrong.

> And if the result is multi-platform but there is no metadata at all, shouldn't metadata be created here for each platform?

Yes but don't know how I can find the target platforms in bridge result. Should I use `ResolveImageConfigOpt`?

Reply:
> Yes but don't know how I can find the target platforms in bridge result.

If `len(res.Refs) > 0` then it is a multi-arch. And platform values are the keys in that map.